examples/aurelia/todos.js

```diff
@@ -0,0 +1,140 @@
+import {ObserverLocator} from 'aurelia-binding';
+import {TodoItem} from './todo-item';
+import _ from 'underscore';
+
+var STORAGE_NAME = 'todomvc-aurelia';
+var ENTER_KEY = 13;
+
+export class Todos {
+	static inject() { return [ObserverLocator]; }
+	constructor(observerLocator, storage = null) {
+		this.items = [];
+		this.filteredItems = [];
+		this.filter = '';
+		this.newTodoTitle = null;
+		this.areAllChecked = false;
+
+		this.observerLocator = observerLocator;
+		this.storage = storage || localStorage;
+		this.load();
+	}
+
+	activate(params) {
+		this.updateFilteredItems(params.filter);
+	}
+
+	onKeyUp(ev) {
+		if (ev.keyCode == ENTER_KEY) {
+			this.addNewTodo(this.newTodoTitle);
+		}
+	}
+
+	addNewTodo(title = this.newTodoTitle) {
+		if (title == undefined) { return; }
+
+		title = title.trim();
+		if (title.length == 0) { return; }
+
+		var newTodoItem = new TodoItem(title);
+		this.observeItem(newTodoItem);
+		this.items.push(newTodoItem);
+		this.newTodoTitle = null;
+		this.updateAreAllCheckedState();
+		this.updateFilteredItems(this.filter);
+		this.save();
+	}
+
+	observeItem(todoItem) {
+		this.observerLocator
+			.getObserver(todoItem, 'title')
+			.subscribe((o, n) => this.onTitleChanged(todoItem));
+
+		this.observerLocator
+			.getObserver(todoItem, 'isCompleted')
+			.subscribe(() => this.onIsCompletedChanged());
+	}
+
+	onTitleChanged(todoItem) {
+		if (todoItem.title == '') {
+			this.deleteTodo(todoItem);
+			this.updateAreAllCheckedState();
+		}
+
+		this.save();
+	}
+
+	onIsCompletedChanged() {
+		this.updateAreAllCheckedState();
+		this.updateFilteredItems(this.filter);
+
+		this.save();
+	}
+
+	deleteTodo(todoItem) {
+		this.items = _(this.items).without(todoItem);
+		this.updateAreAllCheckedState();
+		this.updateFilteredItems(this.filter);
+		this.save();
+	}
+
+	areAllCheckedChanged() {
+		_.each(this.items, i => i.isCompleted = this.areAllChecked);
```

Comment:
a mutation inside of an each is a bit of code smell.. .can we use a map and an assignment?